minionsai/discriminator_only/translator.py

```diff
@@ -57,30 +85,122 @@ def translate(self, game: Game):
+                            legal[i,j] = True
+                    else:
+                        # wrong phase, action is illegal
+                        if game.phase != Phase.MOVE:
+                            continue
+                        # move action
+                        x1, y1 = convert_loc_to_pos(i)
+                        x2, y2 = convert_loc_to_pos(j)
+                        act = MoveAction((x1,y1), (x2,y2))
+                        if game.process_single_move(act, False)[0]:
+                            legal[i,j] = True
+        return legal
+
+    def untranslate_all_actions(self):
+        actions = [[None for i in range(self.num_things)] for j in range(self.num_things)]
+        for i in range(self.num_things):
+            for j in range(self.num_things):
+                if i == self.spawn_pos:
+                    # advance phase action
+                    if j == self.spawn_pos:
+                        actions[i][j] = AdvancePhaseAction()
+                        continue
+                    # spawn action
+                    x2, y2 = convert_loc_to_pos(j)
+                    act = SpawnAction(ZOMBIE, (x2,y2))
+                    actions[i][j] = act
+                else:
+                    # move action
+                    x1, y1 = convert_loc_to_pos(i)
+                    x2, y2 = convert_loc_to_pos(j)
+                    act = MoveAction((x1,y1), (x2,y2))
+                    actions[i][j] = act
+        return actions
+
+    def untranslate_action(self, action):
+        untranslated_action = self.untranslated_actions[action[0]][action[1]]
+        assert not (untranslated_action is None)
+        return untranslated_action
+
+    # def process_action(self, game, action):
+    #     game.process_single_action(action)
+    #     return 1
 
     @staticmethod
     def symmetries(obs):
         # Takes an obs that came out of translator.translate()
         # and returns a list of symmetric observations
         # TODO(david) - this function really should be tested.
+        # print(obs)
+        # print("-----------------------------------")
+        # print(obs['board'])
         board = obs['board']
         num = board.shape[0]
         board = np.reshape(board, [num, BOARD_SIZE, BOARD_SIZE, 3])
         all_boards = [board, np.transpose(board, axes=[0, 2, 1, 3])]
         rotated_boards = [np.flip(np.flip(b, axis=1), axis=2) for b in all_boards]
         all_boards = all_boards + rotated_boards
         all_boards = [np.reshape(b, [num, BOARD_SIZE ** 2, 3]) for b in all_boards]
-        return [{
-            'board': b,
-            'remaining_turns': obs['remaining_turns'],
-            'money': obs['money'],
-            'opp_money': obs['opp_money'],
-            'score_diff': obs['score_diff']
-        } for b in all_boards]
\ No newline at end of file
+
+        new_obs = []
+        for b in all_boards:
+            d = OrderedDict()
```

Comment:
It would be better here to reuse the ordering in `obs`, rather than rewriting it - now there are two sources of truth about the correct order in the codebase.